modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt

```diff
@@ -23,16 +24,20 @@ zephyr_include_directories_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1A ${pdl_dev_cat1
 zephyr_include_directories_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1A ${pdl_dev_cat1a_dir}/include/ip)
 zephyr_library_sources_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1A ${pdl_dev_cat1a_dir}/source/cy_device.c)
 
-
-zephyr_include_directories(${pdl_drv_dir}/include)
 zephyr_include_directories_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1B ${pdl_dev_cat1b_dir}/include)
 zephyr_include_directories_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1B ${pdl_dev_cat1b_dir}/include/ip)
 zephyr_library_sources_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1B ${pdl_dev_cat1b_dir}/source/cy_device.c)
 
+zephyr_include_directories_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1C ${pdl_dev_cat1c_dir}/include)
+zephyr_include_directories_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1C ${pdl_dev_cat1c_dir}/include/ip)
+zephyr_library_sources_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1C ${pdl_dev_cat1c_dir}/source/cy_device.c)
+
 zephyr_library_sources(${pdl_drv_dir}/source/TOOLCHAIN_GCC_ARM/cy_syslib_ext.S)
 
 # Peripheral drivers
-zephyr_library_sources_ifdef(CONFIG_SOC_FAMILY_PSOC6_LEGACY      ${pdl_drv_dir}/source/cy_sysint.c)
+if(CONFIG_SOC_FAMILY_PSOC6_LEGACY OR CONFIG_CPU_CORTEX_M7)
+zephyr_library_sources(${pdl_drv_dir}/source/cy_sysint.c)
```

Comment:
why should cy_sysint.c always be included in every single build for every vendor's m7 CPU?